src/asl.ts

```diff
@@ -3354,16 +3461,89 @@ export namespace ASLGraph {
       })
     );
 
+    /**
+     * Find the updated next value for all of the empty states.
+     * If the updated Next cannot be determined, do not remove the state.
+     */
+    const emptyTransitions = Object.fromEntries(
+      Object.entries(emptyStates).flatMap(([name, { Next }]) => {
+        const getNext = (
+          transition: string,
+          seen: string[] = []
```

Comment:
How long do we expect the chain to get? If it got really long, i'd recommend an immutable Set over an array for `O(lg n)` instead of `O(n)`

Reply:
I had a set, but changed to array to simplify. I don't expect it to get long, the array is reset for each state. Maybe 3-8 state names max. Average 0-1.